cmake complaining about cyclic dependencies

**OS:** Ubuntu
**Version:** Trusty
**Commit:** b436ac9bc6c56d4760ab1b48c2762b7081c7ff45

I'm having troubles getting OpenRTC2 to build. I've installed all dependencies (libpng16 and jansson 2.7 by source), now cmake is complaining about this:


$ cmake .
-- Configuring done
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "linkable_sections" of type UTILITY
    depends on "segfiles" (strong)
  "openrct2" of type EXECUTABLE
    depends on "linkable_sections" (strong)
    depends on "segfiles" (strong)
  "segfiles" of type UTILITY
    depends on "openrct2" (strong)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
-- Build files have been written to: /home/jglauche/src/OpenRCT2

It says that it has written build files but it doesn't produce any build files. 

I have tried option static both ON and OFF. It doesn't make a difference. 

Any ideas to fix that?

Actually, you were right. It's more of a hassle than I thought.

First, it seems that cmake isn't set up to figure all out requirements because it forces you to cross compile it.

- I had to install libc6-dev-i386.
- I had to install zlib1g-dev:i386, which basically removed all my build system (reinstalled build-essential and g++-4.8-multilib

Next, the 64-bit dev package of curl is acting up. 

 Building CXX object CMakeFiles/openrct2.dir/src/network/http.cpp.o
In file included from /usr/include/curl/curl.h:35:0,
                 from /home/jglauche/src/OpenRCT2/src/network/http.cpp:21:
/usr/include/curl/curlrules.h:143:41: error: size of array ‘__curl_rule_01__’ is negative
     [CurlchkszEQ(long, CURL_SIZEOF_LONG)];
                                         ^
/usr/include/curl/curlrules.h:153:53: error: size of array ‘__curl_rule_02__’ is negative
     [CurlchkszEQ(curl_off_t, CURL_SIZEOF_CURL_OFF_T)];

replacing it by the i386 package of libcurl4-openssl-dev fixed that.

Building works so far, the linker now complains that my self-build libraries are not compatible. Yes, I know. Gotta compile those for 32 bit.

Are there any big hurdles in compiling the whole thing in 64 bit..?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,6 +23,10 @@ set (ORCT2_RESOURCE_DIR ${CMAKE_INSTALL_PREFIX}/share/${PROJECT}/)
 
 project(${PROJECT})
 
+if (CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
+	message(FATAL_ERROR "Building in-source is not supported! Create a build dir and remove ${CMAKE_SOURCE_DIR}/CMakeCache.txt")
+endif()
+
 add_definitions(-DORCT2_RESOURCE_DIR="${ORCT2_RESOURCE_DIR}")
 add_definitions(-DHAVE_CONFIG_H)
 add_definitions(-DCURL_STATICLIB)

